File: api/internal/core/dto/workspace.go
package dto

type CreateWorkspaceInput struct {
	Name  string `json:"name" binding:"required,max=80"`
	Icon  string `json:"icon" binding:"max=32"`
	Color string `json:"color" binding:"max=32"`
}

type UpdateWorkspaceInput struct {
	Name      string `json:"name" binding:"required,max=80"`
	Icon      string `json:"icon" binding:"max=32"`
	Color     string `json:"color" binding:"max=32"`
	Collapsed bool   `json:"collapsed"`
}

type ReorderWorkspaceInput struct {
	IDs []string `json:"ids" binding:"required,min=1"`
}

File: website/src/entities/workspace.ts
export type Workspace = {
  id: string
  name: string
  icon: string
  color: string
  position: number
  collapsed: boolean
  createdAt: string
  updatedAt: string
}

export type CreateWorkspaceInput = Pick<Workspace, 'name' | 'icon' | 'color'>
export type UpdateWorkspaceInput = Pick<Workspace, 'name' | 'icon' | 'color' | 'collapsed'>
